homeassistant/components/google_assistant/smart_home.py

```diff
@@ -144,6 +160,37 @@ def celsius(deg: Optional[float]) -> Optional[float]:
         if deg is None:
             return None
         return round(METRIC_SYSTEM.temperature(deg, units.temperature_unit), 1)
+
+    if entity.domain == sensor.DOMAIN:
+        google_domain = entity.attributes.get(ATTR_GOOGLE_ASSISTANT_TYPE)
+        if google_domain == climate.DOMAIN:
```

Comment:
@philk, I totally agree with u on ERROR_NOT_SUPPORTED and I don't wanna "(git blame)[https://github.com/home-assistant/home-assistant/pull/9632/files#diff-1fb7fb9d15a7debaad124e1e793ff0b1R104]" anyone, but how the hell we ended up with that code which indeed reports brightness without even checking for if it's a supported feature or trait of the device.

I'll try handle my case here no pb, but don't u think that we should handle that initial commit with a different PR and real fix for that, since that can happen for anyone if the they don't return immediately a result and their code slips into that dangerous section ?